### PR TITLE
Enable mpv2 jobs

### PR DESCRIPTION
**Pull request by yaakovi**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
restored the marketplace V2 jobs in the `on-push` and `upload` flows

### CODE REVIEW

File: .gitlab/ci/on-push.yml [hunk 1]
```diff
@@ -101,49 +101,50 @@ create-instances:
...
-#     - section_end "Collect Tests and Content Packs"
-
-#     - section_start "Prepare Content Packs for Testing"
-#     - ./Tests/scripts/prepare_content_packs_for_testing.sh "$GCS_MARKET_V2_BUCKET" "$STORAGE_BASE_PATH" "marketplacev2"
-#     - section_end "Prepare Content Packs for Testing"
-
-#     - section_start "Upload Artifacts to GCP" --collapsed
-#     - ./Tests/scripts/upload_artifacts.sh
-#     - section_end "Upload Artifacts to GCP"
+create-instances-mpv2:
```
Comment (avidan-H) [inline .gitlab/ci/on-push.yml:104]: Same here - rename to be more accurate, please.

File: .gitlab/ci/bucket-upload.yml [hunk 2]
```diff
@@ -54,17 +54,17 @@ create-instances-upload-flow:
...
-#    - create-instances-mpv2
-#  variables:
-#    IFRA_ENV_TYPE: "Bucket-Upload"
-#    ARTIFACTS_FOLDER: "${ARTIFACTS_FOLDER_MPV2}"
-#  rules:
-#    - if: '$CI_COMMIT_BRANCH =~ /pull\/[0-9]+/'
-#      when: never
-#    - if: '$BUCKET_UPLOAD == "true"'
-#    - if: '$FORCE_BUCKET_UPLOAD == "true"'
+create-instances-mpv2-upload-flow:
```
Comment (avidan-H) [inline .gitlab/ci/bucket-upload.yml:57]: Let's rename the job to be a more accurate description.

File: .gitlab/ci/bucket-upload.yml [hunk 3]
```diff
@@ -54,17 +54,17 @@ create-instances-upload-flow:
...
+ extends:
+   - create-instances-mpv2
+ variables:
+   IFRA_ENV_TYPE: "Bucket-Upload"
+   ARTIFACTS_FOLDER: "${ARTIFACTS_FOLDER_MPV2}"
+ rules:
+   - if: '$CI_COMMIT_BRANCH =~ /pull\/[0-9]+/'
+     when: never
+   - if: '$BUCKET_UPLOAD == "true"'
+   - if: '$FORCE_BUCKET_UPLOAD == "true"'
```
Comment (avidan-H) [inline .gitlab/ci/bucket-upload.yml:69]: Let's add a commented-out placeholder for a job to test the successful installation of mpv2 packs to instances (which you can install mpv2 packs on).

mahsu kmo,
```
# TO-DO: Job to test installation of mpv2 packs - issue <link-here>
# install-mpv2-pack-on-mpv2-instances
```

**Review by avidan-H — CHANGES_REQUESTED** [review]
Per your own comment, let's leave the `id_set.json` filename the same since they will sit in different directories and/or buckets.

**Review by avidan-H — APPROVED** [review]

File: .gitlab/ci/bucket-upload.yml [hunk 4]
```diff
@@ -199,6 +205,65 @@ upload-packs-to-marketplace:
...
+upload-packs-to-marketplace-v2:
+  tags:
+    - gke
+  needs: ["run-validations-upload-flow", "run-unittests-and-lint-upload-flow", "prepare-testing-bucket-mpv2-upload-flow"]
+  stage: upload-to-marketplace
+  artifacts:
+    expire_in: 48 hrs
+    paths:
+      - /builds/xsoar/content/artifacts/*
+    when: always
```
Comment (esharf) [inline .gitlab/ci/bucket-upload.yml:217]: Let’s open an issue to remove it once mp-v2 will be ready

File: .gitlab/ci/on-push.yml [hunk 5]
```diff
@@ -101,49 +101,50 @@ create-instances:
...
-#     - ./Tests/scripts/upload_artifacts.sh
-#     - section_end "Upload Artifacts to GCP"
+prepare-testing-bucket-mpv2:
+  tags:
+    - gke
+  extends:
+    - .default-job-settings
+  rules:
+    - if: '$CI_PIPELINE_SOURCE =~ /^(push|contrib)$/'
+    - if: '$NIGHTLY'
```
Comment (esharf) [inline .gitlab/ci/on-push.yml:111]:
```suggestion
```
we don’t need to do this in the nightly build.
currently, the nightly installs from the production bucket and use manual pack upload for the tests.
so I think that the behavior should be the same.

File: .gitlab/ci/on-push.yml [hunk 6]
```diff
@@ -101,49 +101,50 @@ create-instances:
...
+  stage: create-instances
+  script:
+    - !reference [.download-demisto-conf]
+    - !reference [.create-id-set-mp-v2]
+    - !reference [.create-release-notes-and-common-docs]
+    - echo "$ARTIFACTS_FOLDER"
+
+    - section_start "Create Content Artifacts and Update Conf" --collapsed
+    - demisto-sdk create-content-artifacts -a $ARTIFACTS_FOLDER --cpus 8 --content_version $CONTENT_VERSION --marketplace "marketplacev2" -fbi -idp ./Tests/id_set.json >> $ARTIFACTS_FOLDER/logs/create_content_artifacts.log
+    - zip -j $ARTIFACTS_FOLDER/uploadable_packs_mpv2.zip $ARTIFACTS_FOLDER/uploadable_packs/* >> $ARTIFACTS_FOLDER/logs/zipping_uploadable_packs.log || ((($? > 0)) && echo "failed to zip the uploadable packs, ignoring the failure")
```
Comment (esharf) [inline .gitlab/ci/on-push.yml:126]:
```suggestion
    - zip -j $ARTIFACTS_FOLDER/uploadable_packs_mpv2.zip $ARTIFACTS_FOLDER/uploadable_packs/* >> $ARTIFACTS_FOLDER/logs/zipping_uploadable_packs.log || echo "failed to zip the uploadable packs, ignoring the failure"
```
should be enough.

**Review by esharf — CHANGES_REQUESTED** [review]
you changed some Release Notes files, (just making sure that it’s Intentionally).

File: .gitlab/ci/bucket-upload.yml (hunk 4 above)
Comment (yaakovi) [reply .gitlab/ci/bucket-upload.yml:217]: remove what?

File: .gitlab/ci/on-push.yml (hunk 5 above)
Comment (yaakovi) [reply .gitlab/ci/on-push.yml:111]: it is mostly done for testing the flow, even if we don't use the resulting bucket.

File: .gitlab/ci/on-push.yml [hunk 7]
```diff
@@ -101,49 +101,50 @@ create-instances:
...
+    policy: pull-push
+  variables:
+    ARTIFACTS_FOLDER: "${ARTIFACTS_FOLDER_MPV2}"
+  needs: []
+  stage: create-instances
+  script:
+    - !reference [.download-demisto-conf]
+    - !reference [.create-id-set-mp-v2]
+    - !reference [.create-release-notes-and-common-docs]
+    - echo "$ARTIFACTS_FOLDER"
```
Comment (Noy-Maimon) [inline .gitlab/ci/on-push.yml:122]: same here, not sure if needed, if so the we should add a description for what we are printing

**Review by Noy-Maimon — APPROVED** [review]

File: .gitlab/ci/bucket-upload.yml (hunk 4 above)
Comment (yaakovi) [reply .gitlab/ci/bucket-upload.yml:217]: We are saving the artifacts in the artifact bucket, so we don't need to keep them for an extended period in Gitlab.

File: Tests/scripts/gitlab_slack_notifier.py [hunk 8]
```diff
@@ -12,37 +12,35 @@
...
+
+    parser.add_argument('-s', '--slack_token', help='The token for slack', required=True)
     parser.add_argument(
         '-ch', '--slack_channel', help='The slack channel in which to send the notification', default=CONTENT_CHANNEL
     )
-    parser.add_argument('-gp', '--gitlab_project_id', help='The gitlab project id', default=GITLAB_PROJECT_ID)
-    parser.add_argument(
-        '-tw', '--triggering-workflow', help='The type of ci pipeline workflow the notifier is reporting on',
-        choices=WORKFLOW_TYPES)
+
```
Comment (avidan-H) [inline Tests/scripts/gitlab_slack_notifier.py:43]: Why the reordering? You are effectively only removing the `env_results_path` argument.

Comment (yaakovi) [reply Tests/scripts/gitlab_slack_notifier.py:43]: I changed the arguments, so I separated them into groups.
I reverted most of them, and this is the result - I will revert to the original state.

**Pull request merged**